### PR TITLE
Add error message when none is provided

### PR DESCRIPTION
**Pull request by mayerm94**
When the execution does not provide an error message, Bonnie 500s and chalks the error up to network connectivity issues. What the error message should be set to on line 12 is up for discussion.

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1342
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered. 
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: NA
- [x] Justification for using JIRA tests: NA
- [x] JIRA tests have been added to sprint NA

**Reviewer 1:**

Name: @hossenlopp
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA tests have been run and pass N/A
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree N/A

**Reviewer 2:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: test/unit/helpers/error_helper_test.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      error_info = {section: 'cql-measure-calculation',
+                    cms_id: 'CMSv0', type: 'javascript',
+                    url: 'http://127.0.0.1:3000/#measures/21C422D4-1658-478D-AEC2-B87B73638C41/patients/new',
+                    controller: 'application', action: 'client_error'}
+      exception = {section: 'cql-measure-calculation', cms_id: 'CMSv0', type: 'javascript', 
+                    url: 'http://127.0.0.1:3000/#measures/21C422D4-1658-478D-AEC2-B87B73638C41/patients/new', 
+                    controller: 'application', action: 'client_error', msg: 'No error message provided'}
+      request = "empty"
+      error_message = ErrorHelper.describe_error(error_info, exception, request)
+      assert_equal error_message[:body], "One of the data elements associated with the measure is causing an issue. Please review the elements associated with the measure to verify that they are all constructed properly.<br>Error message: <b>No error message provided</b>"
```
Comment (hossenlopp) [inline test/unit/helpers/error_helper_test.rb]: The expected value should come first in `assert_equal` usage.

File: app/helpers/error_helper.rb [hunk 2]
```diff
@@ -5,11 +5,11 @@ module ErrorHelper
   # not be categorized, no error message will be shown (but an email
   # will still be sent to the development team, if on production).
   def self.describe_error(error_info, exception, request)
+    error_info[:msg] = "An unspecified error has occurred." unless error_info[:msg]
     # Do not process errors if their message includes Costanza. These are
     # errors passed up to Thorax.onException from Costanza, which we do
     # not care about (we've already handled them).
     return if error_info[:msg].include? 'Costanza'
-
```
Comment (holmesie) [inline app/helpers/error_helper.rb]: Why delete this line? Was this because of a rubocop complaint?

File: app/helpers/error_helper.rb [hunk 3]
```diff
@@ -5,11 +5,11 @@ module ErrorHelper
   # not be categorized, no error message will be shown (but an email
   # will still be sent to the development team, if on production).
   def self.describe_error(error_info, exception, request)
+    error_info[:msg] = "An unspecified error has occurred." unless error_info[:msg]
```
Comment (holmesie) [inline app/helpers/error_helper.rb]: Even though this is somewhat straight forward, every action in this helper is well commented. I'd write a comment here for this line about why it is necessary.

Comment (holmesie) [reply app/helpers/error_helper.rb]: Also, I'm surprised rubocop did not complain about the use of double quotes over single for this.

File: app/helpers/error_helper.rb [hunk 4]
```diff
@@ -5,6 +5,9 @@ module ErrorHelper
   # not be categorized, no error message will be shown (but an email
   # will still be sent to the development team, if on production).
   def self.describe_error(error_info, exception, request)
+    # Set the error message if it was not previously set
+    error_info[:msg] = 'An unspecified error has occurred.' unless error_info[:msg]
```
Comment (hossenlopp) [inline app/helpers/error_helper.rb]: This doesn't handle the case of if `error_info[:msg]` is `""`. Should we take care of that, or not?

**Review by hossenlopp — COMMENTED** [review]
Doesn't handle situation of an error throw in a costanza run block that does have a message.

For example, run this in the console:
```
Costanza.run('cql-measure-calculation', { cms_id: 'CMS72v6'}, function() { throw new Error() })
```
Costanza sends `""` for `:msg`.

**Review by hossenlopp — APPROVED** [review]

**Review by jbradl11 — APPROVED** [review]
Code looks good. Nice error message.

**Pull request closed without merging**